### PR TITLE
draft readme file for defrag sample

### PR DESCRIPTION
**Pull request by d45**
@tsjoblad @jonas-eckhardt 

I've added some info to a readme for the defrag sample. 
It's pretty slim on content, but it might be enough to point them in the right direction.

### CODE REVIEW

File: Community-Supported/defragment-hyper-file/README.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+For a description of how fragmentation can occur in the `.hyper` file and for ways of minimizing it's occurrence, see [Optimize Hyper File Storage](https://help.tableau.com/current/api/hyper_api/en-us/docs/hyper_api_defrag.html).
+
+# Get started
+
+## __Prerequisites__
+
+To run the script, you will need:
+
+- a computer running Windows or macOS
```
Comment (jonas-eckhardt) [inline Community-Supported/defragment-hyper-file/README.md:22]: Linux?

File: Community-Supported/defragment-hyper-file/README.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+- Tableau Desktop v10.5 or higher
+
+- Python 3.7
```
Comment (jonas-eckhardt) [inline Community-Supported/defragment-hyper-file/README.md:26]: 3.6?

File: Community-Supported/defragment-hyper-file/README.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+## Run the sample
+
+As a quick test of how the sample works, you don't need to make any changes to the `defragment_data_of_existing_hyper_file.py` file. You can simply run the sample on a `.hyper` file to see how it can reduce fragmentation. The Python sample reads an existing `.hyper` file and copies all the tables and data into a new file.
+
+Ensure that you have installed the requirements and execute the Python file. 
+
+1. The following instructions assume that you have set up a virtual environment for Python. For more information on creating virtual environments, see [venv - Creation of virtual environments](https://docs.python.org/3/library/venv.html){:target="_blank"}{:ref="noopener"} in the Python Standard Library.
+
+1. Open a terminal and navigate to the `venv` directory.
```
Comment (jonas-eckhardt) [inline Community-Supported/defragment-hyper-file/README.md:36]: We should add one step to activate the virtual environment

File: Community-Supported/defragment-hyper-file/README.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    ```cli
+    (venv)c:\mydir> python defragment_data_of_existing_hyper_file.py -o Newfile.hyper Oldfile.hyper
+    Successfully converted table "input_database"."Extract"."Extract"
+    Successfully converted Oldfile.hyper into Newfile.hyper
+   ```
+
+
+## __Customization__
+
+The script, `defragment_data_of_existing_hyper_file.py`, should work in most cases. However, if your Hyper file contains metadata that is not stored in the table definitions, for example, column descriptions or assumed constraints, you might need to modify the script to preserve the metadata. If you created the constraints and definitions yourself, using your own SQL statements, the Python script should give you a good starting point for adding your own code.
```
Comment (jonas-eckhardt) [inline Community-Supported/defragment-hyper-file/README.md:56]: I would not formulate it that way. The script works, however, it does not copy over the (assumed) constraints.

File: Community-Supported/defragment-hyper-file/README.md (hunk 1 above)
Comment (hofst) [reply Community-Supported/defragment-hyper-file/README.md:22]: maybe just reference the requirements from the HyperAPI docs

**Comment by tsjoblad** [comment]
Manually merged, closing PR.

**Pull request closed without merging**